src/Metadata.php

```diff
@@ -169,4 +175,61 @@ protected function getPlaceholderIllustration()
     {
         return 'app/img/logo.png';
     }
+
+    /**
+     * @param $file
+     *
+     * @return array
+     */
+    protected function getEntriesFromCSV($file)
+    {
+        return Reader::createFromPath($file)->fetchAssoc(0);
+    }
+
+    /**
+     * @param $file
+     *
+     * @return array
+     */
+    protected function getEntriesFromCache($file)
+    {
+        $identifier = $this->getCacheIdentifier($file);
+        $lastModifiedAt = (new SplFileInfo($file))->getMTime();
+
+        if ($cached = $this->cache->get($identifier)) {
+            if (!$this->isModified($cached, $lastModifiedAt)) {
+                return $cached['meta'];
+            }
+        };
+
+        $entries = $this->getEntriesFromCSV($file);
+
+        $this->cache->forever($identifier, [
+            'last_modified_at' => $lastModifiedAt,
+            'meta' => $entries
+        ]);
```

Comment:
Not file/array but considering your issue seems to be for production/staging environments we shouldn't use those cache drivers there anyway if performance is the issue so it'd make sense that these two'd go together